common/api/core-transformer.api.md

```diff
@@ -26,6 +26,14 @@ import { RelationshipProps } from '@itwin/core-backend';
 import { Schema } from '@itwin/ecschema-metadata';
 import { SchemaKey } from '@itwin/ecschema-metadata';
 
+// @beta
+export enum DeadPredecessorsBehavior {
```

Comment:
Did you consider a union of strings instead of an enum?
```
  const opts = { deadPredecessorBehavior: "reject" };
```
vs
```
  import { DeadPredecessorBehavior } from "@itwin/core-transformer";
  const opts = { deadPredecessorBehavior: DeadPredecessorBehavior.Reject };
```

Reply:
I did.
I was hoping that typescript would allow us to pass strings as enum arguments to remove the need for the importing,
but it doesn't (to prevent the value being changed from underneath I suppose).
I prefer enums generally because I think it's easier to search for references to values, but I agree it's clunky for users.

What do you think about the following custom-enum-like object? I don't love it because it's not really idiomatic but it fits all of the above much better imo:

```ts
const DeadPredecessorBehavior = {
  Reject: "reject",
  Ignore: "ignore",
} as const;

// in utility types or provided from elsewhere
type ValueOf<T> = T[keyof T];

interace Options {
  deadPredecessorBehavior: ValueOf<DeadPredecessorBehaviors>;
}
```
now you can pass a string or `DeadPredecessorBehavior.OPTION`

Reply:
> I prefer enums generally because I think it's easier to search for references to values, but I agree it's clunky for users.

The library is for users. Design APIs accordingly.

> What do you think about the following custom-enum-like object?

Nope.

Reply:
I agree will switch it to a string enum

Reply:
please take a look at the string-enum and resolve if there aren't any other issues.